controllers/authpolicy_controller_test.go

```diff
@@ -132,6 +129,52 @@ var _ = Describe("AuthPolicy controller", func() {
 			Expect(authConfig.Spec.Conditions[0].Any[0].Any[0].All[1].Value).To(Equal("/toy.*"))
 		})
 
+		It("Attaches policy to a Gateway with hostname in listeners", func() {
+			gatewayName := fmt.Sprintf("%s-with-hostnames", testGatewayName)
+			gateway := testBuildBasicGateway(gatewayName, testNamespace)
+			Expect(gateway.Spec.Listeners).To(HaveLen(1))
+			// Set hostname
+			gateway.Spec.Listeners[0].Hostname = &[]gatewayapiv1.Hostname{"*.example.com"}[0]
+			err := k8sClient.Create(context.Background(), gateway)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(testGatewayIsReady(gateway), 15*time.Second, 5*time.Second).Should(BeTrue())
+
+			policy := &api.AuthPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "gw-auth",
+					Namespace: testNamespace,
+				},
+				Spec: api.AuthPolicySpec{
+					TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+						Group:     "gateway.networking.k8s.io",
+						Kind:      "Gateway",
+						Name:      gatewayapiv1.ObjectName(gatewayName),
+						Namespace: ptr.To(gatewayapiv1.Namespace(testNamespace)),
+					},
+					AuthScheme: testBasicAuthScheme(),
+				},
+			}
+
+			err = k8sClient.Create(context.Background(), policy)
+			logf.Log.V(1).Info("Creating AuthPolicy", "key", client.ObjectKeyFromObject(policy).String(), "error", err)
+			Expect(err).ToNot(HaveOccurred())
+
+			// check policy status
+			Eventually(testPolicyIsReady(policy), 60*time.Second, 5*time.Second).Should(BeTrue())
```

Comment:
Mhmm, maybe we can't rely on the time frame provided

Reply:
the check is correct. It was rightfully saying something is wrong. 

I missed adding a route. So a policy targeting a gateway without routes do not have any effect and the policy is not ready.